Second_semestr/Homework_4/Parser/Parser/ParseTree.cs

```diff
@@ -0,0 +1,197 @@
+        public int Calculate()
+        {
+            return this.root.Calculate();
+        }
+
+        private Node root;
+
+        /// <summary>
+        /// returns token(operands or operators(+, -, /, *)) which is located after the "position"
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <param name="position"></param>
+        /// <returns></returns>
+        private string Token(string expression, ref int position)
+        {
+            var length = expression.Length;
+            var token = "";
+            for (var i = position; i < length; ++i)
+            {
+                var symbol = expression[i];
+                if ((expression[i] == ' ') || (expression[i] == '(') || (expression[i] == ')'))
+                {
+                    ++position;
+                    continue;
+                }
+
+                if ((expression[i] >= '0') && (expression[i] <= '9'))
+                {
+                    token += expression[i];
+                    for (var j = i + 1; j < length; ++j)
+                    {
+                        if ((expression[j] < '0') || (expression[j] > '9'))
+                        {
+                            break;
+                        }
+
+                        ++position;
+                        token += expression[j];
+                    }
+
+                    ++position;
+                    return token;
+                }
+
+                if ((expression[i] == '+') || (expression[i] == '*') || (expression[i] == '/') || (expression[i] == '-'))
+                {
+                    ++position;
+                    return token + expression[i];
+                }
+            }
+
+            return "error";
+        }
+
+
+        /// <summary>
+        /// creates new node and makes it left or right child of the given node
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="expression"></param>
+        /// <param name="position"></param>
+        private void CreateNode(Node node, string expression, ref int position)
+        {
+            if ((node.Left != null) && (node.Right != null))
+            {
+                return;
+            }
+
+            var token = Token(expression, ref position);
+            Node newNode;
+            switch (token[0])
+            {
+                case '+':
+                    {
+                        newNode = new Addition();
+                        break;
+                    }
+
+                case '-':
+                    {
+                        newNode = new Subtraction();
+                        break;
+                    }
+
+                case '*':
+                    {
+                        newNode = new Multiplication();
+                        break;
+                    }
+
+                case '/':
+                    {
+                        newNode = new Division();
+                        break;
+                    }
```

Comment:
видели в смысле у кого то так же?

2016-05-16 23:47 GMT+03:00 Альберт Мухаммадиев <beehead02@gmail.com>:

> точно, можно же просто не писать метод)
>
> 2016-05-16 18:55 GMT+03:00 Yurii Litvinov <notifications@github.com>:
>
>> In Second_semestr/Homework_4/Parser/Parser/ParseTree.cs
>> <https://github.com/AlbertMukhammadiev/University/pull/31#discussion_r63376675>
>> :
>>
>> > +                    {
>> > +                        newNode = new Subtraction();
>> > +                        break;
>> > +                    }
>> > +
>> > +                case '*':
>> > +                    {
>> > +                        newNode = new Multiplication();
>> > +                        break;
>> > +                    }
>> > +
>> > +                case '/':
>> > +                    {
>> > +                        newNode = new Division();
>> > +                        break;
>> > +                    }
>>
>> что-то я такой код уже только что видел
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/AlbertMukhammadiev/University/pull/31/files/5d30c1adfb21110b4e429cc5d8729924cc7563e2#r63376675>
>>
>
>